research/query_service/gremlin/scripts/run_gaia.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Alibaba Group Holding Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import tempfile
+import os
+import subprocess
+from utils import get_hosts_from_toml, hosts_dict_to_list, sync_process, clean_up, str2bool
+
+def extract_file_name(line: str):
+    """
+    Extract the file name from a line getting by a line of string that contains a directory like `/path/to/file`,
+    where `file` will be extracted.
+    :param line: A line of string
+    :return: The extracted tuple of absolute path and filename
+    """
+    start_idx = line.rfind('hdfs://')
+    if start_idx == -1:
+        absolute_path = line
+    else:
+        absolute_path = line[start_idx:]
+    idx = absolute_path.rfind('/')
+    filename = absolute_path[idx + 1:]
+    if filename.startswith('.'):
+        return None, None
+    else:
+        return absolute_path, filename
+
+
+####################################################################
+####################### Prepare the arguments ######################
+####################################################################
+
+TMP_FOLDER = "tmp4p1wlx9p"
```

Comment:
Use `with tempfile.TemporaryFile(): ...` instead.

Reply:
This is fine. As the folder is a temporary one.  I change the file name to "gaia_tmp" to avoid mis-understanding